src/autopas/LogicHandler.h

```diff
@@ -560,16 +557,56 @@ class LogicHandler {
+   *
+   * @param boxLength Size per dimension for the box that should be covered by locks (should include halo).
+   * @param interactionLengthInv Inverse of the side length of the virtual boxes one lock is responsible for.
+   *
    */
   void initSpacialLocks(const std::array<double, 3> &boxLength, double interactionLengthInv) {
     using namespace autopas::utils::ArrayMath::literals;
     using autopas::utils::ArrayMath::ceil;
     using autopas::utils::ArrayUtils::static_cast_copy_array;
 
-    // one lock per interaction length + one for each halo region
-    const auto locksPerDim = static_cast_copy_array<size_t>(ceil(boxLength * interactionLengthInv) + 2.);
+    // The maximum number of spatial locks is capped at 1e6.
+    // Without this cap, very large domains (or tiny cutoffs) would generate an insane number of locks,
+    // that could blow up the memory.
+    constexpr size_t maxNumSpacialLocks{1000000};
+
+    // One lock per interaction length or less if this would generate too many.
+    const auto locksPerDim = [&]() {
+      // First naively calculate the number of locks if we simply take the desired cell length.
+      // Ceil because both decisions are possible, and we are generous gods.
+      const std::array<size_t, 3> locksPerDimNaive =
+          static_cast_copy_array<size_t>(ceil(boxLength * interactionLengthInv));
+      const auto totalLocksNaive =
+          std::accumulate(locksPerDimNaive.begin(), locksPerDimNaive.end(), 1ul, std::multiplies<>());
+      // If the number of locks is within the limits everything is fine and we can return.
+      if (totalLocksNaive <= maxNumSpacialLocks) {
+        return locksPerDimNaive;
+      } else {
+        // If the number of locks grows too large, calculate the locks per dimension proportionally to the side lengths.
+        // Calculate side length relative to dimension 0.
+        const std::array<double, 3> boxSideProportions = {
+            1.,
+            boxLength[0] / boxLength[1],
+            boxLength[0] / boxLength[2],
+        };
+        // With this, calculate the number of locks the first dimension should receive.
+        const auto prodProportions =
+            std::accumulate(boxSideProportions.begin(), boxSideProportions.end(), 1., std::multiplies<>());
+        // Needs floor, otherwise we exceed the limit.
+        const auto locksInFirstDimFloat = std::floor(std::cbrt(maxNumSpacialLocks * prodProportions));
+        // From this and the proportions relative to the first dimension, we can calculate the remaining number of locks
+        const std::array<size_t, 3> locksPerDimLimited = {
+            static_cast<size_t>(locksInFirstDimFloat),  // omitted div by 1
+            static_cast<size_t>(locksInFirstDimFloat / boxSideProportions[1]),
+            static_cast<size_t>(locksInFirstDimFloat / boxSideProportions[2]),
+        };
+        return locksPerDimLimited;
+      }
+    }();
```

Comment:
Given that this lambda is called once and directly, it should probably be highlighted in some way/ or not be a lambda anymore.

Three suggestions to refactor to avoid confusion:
1) Make it a proper static private function of the class calculating the `limitedLocksPerDim`
2) Unfold the lambda, place the body directly into the surrounding function
3) Change the initializer from `const auto locksPerDim` to `const std::array<size_t, 3> locksPerDim`. This way, a developer sees in the **first** line of the declaration that this is not a lambda but just a complex initialization.

I would go with either 1) or 3)